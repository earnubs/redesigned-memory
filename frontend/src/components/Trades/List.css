.eb-Trades-List {
  width: 100%;
}

.eb-Trades-List td {
  font-family: "Roboto Condensed", sans-serif;
}

table, tr, td, th {
  margin: 0;
  padding: 0;
  border: none;
}

table {
  border-collapse: collapse;
}

.eb-TradesList-row td {
  padding:8px 16px;
  border-bottom: 1px solid #dde2e3;
}

.eb-TradesList-head {
  text-transform: uppercase;
  color: #84959a;
}

.eb-TradesList-head th {
  font-family: "Roboto", sans-serif;
  font-size: 0.875rem;
  font-weight: 400;
  text-align: left;
  padding:24px 16px 8px 16px;
  border-bottom: 1px solid #dde2e3;
}

.eb-TradesList--lighter {
  color: #84959a;
}
