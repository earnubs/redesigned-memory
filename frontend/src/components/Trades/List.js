import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

import './List.css';

export const NUMBER_FORMAT = '0,0.00';

export function TableHead() {
  return (
    <thead>
      <tr className="eb-TradesList-head">
        <th>
            Sell CCY
        </th>
        <th>
            Sell amount
        </th>
        <th>
            Buy CCY
        </th>
        <th>
            Buy amount
        </th>
        <th>
            Rate
        </th>
        <th>
            Date booked
        </th>
      </tr>
    </thead>
  );
}

export function TableBody({ trades }) {
  return (
    <tbody>
      {
        trades.length ?
          trades.map(trade => (
            <tr key={trade.id} className="eb-TradesList-row">
              <td className="eb-TradesList--lighter">
                {trade.sell_currency}
              </td>
              <td>
                {numeral(trade.sell_amount).format(NUMBER_FORMAT)}
              </td>
              <td className="eb-TradesList--lighter">
                {trade.buy_currency}
              </td>
              <td>
                {numeral(trade.buy_amount).format(NUMBER_FORMAT)}
              </td>
              <td>
                {trade.rate}
              </td>
              <td>
                { trade.date_booked }
              </td>
            </tr>
          )) : null
      }
    </tbody>
  );
}

TableBody.defaultProps = {
  trades: [],
};

TableBody.propTypes = {
  trades: PropTypes.array,
};

export function List({ trades }) {
  return (
    <div>
      <h2>Booked Trades</h2>
      <Link to="/new">[plus icon] New Trade</Link>
      <table className="eb-Trades-List">
        <TableHead />
        <TableBody trades={ trades } />
      </table>
    </div>
  );
}

List.propTypes = {
  trades: PropTypes.array.isRequired,
};

export default List;
