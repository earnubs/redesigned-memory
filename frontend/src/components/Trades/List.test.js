import React from 'react';
import { shallow } from 'enzyme';

import List, { TableHead, TableBody } from './List';

import { trades } from './index.test.js';

describe('List components', () => {
  it('should render TableHead as expected', function() {
    const wrapper = shallow(<TableHead />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render TableBody as expected', function() {
    const wrapper = shallow(<TableBody trades={trades} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render List as expected', function() {
    const wrapper = shallow(<List trades={trades} />);

    expect(wrapper).toMatchSnapshot();
  });
});
