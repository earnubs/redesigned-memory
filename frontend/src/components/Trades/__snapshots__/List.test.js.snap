// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List components should render List as expected 1`] = `
<div>
  <h2>
    Booked Trades
  </h2>
  <Link
    to="/new"
  >
    [plus icon] New Trade
  </Link>
  <table
    className="eb-Trades-List"
  >
    <TableHead />
    <TableBody
      trades={
        Array [
          Object {
            "buy_amount": "87.81",
            "buy_currency": "GBP",
            "date_booked": "2019-06-10T20:06:57.313042Z",
            "id": "TR3W659X2",
            "rate": "0.878085",
            "sell_amount": "100.00",
            "sell_currency": "USD",
          },
        ]
      }
    />
  </table>
</div>
`;

exports[`List components should render TableBody as expected 1`] = `
<tbody>
  <tr
    className="eb-TradesList-row"
    key="TR3W659X2"
  >
    <td
      className="eb-TradesList--lighter"
    >
      USD
    </td>
    <td>
      100.00
    </td>
    <td
      className="eb-TradesList--lighter"
    >
      GBP
    </td>
    <td>
      87.81
    </td>
    <td>
      0.878085
    </td>
    <td>
      10/06/2019 09:06
    </td>
  </tr>
</tbody>
`;

exports[`List components should render TableHead as expected 1`] = `
<thead>
  <tr
    className="eb-TradesList-head"
  >
    <th>
      Sell CCY
    </th>
    <th>
      Sell amount
    </th>
    <th>
      Buy CCY
    </th>
    <th>
      Buy amount
    </th>
    <th>
      Rate
    </th>
    <th>
      Date booked
    </th>
  </tr>
</thead>
`;
